h5py/tests/test_dataset.py

```diff
@@ -1247,7 +1247,7 @@ def test_int(self):
         self.assertArrayEqual(ds[1], np.arange(0))
         self.assertArrayEqual(ds[2], np.array([1, 2, 3]))
         self.assertArrayEqual(ds[1], np.arange(0))
-        ds[0:2] = np.array([np.arange(5), np.arange(4)])
+        ds[0:2] = np.array([np.arange(5), np.arange(4)], dtype=object)
```

Comment:
These "ragged arrays" will emit a warning for the upcoming NumPy 1.19 release since they cannot be turned into arrays of numerics.